app/loghighlighter.cpp

```diff
@@ -61,15 +61,12 @@ void LogHighlighter::highlightBlock(const QString &text)
 {
     // Try each highlighting pattern and apply formatting if it matches
     for (const auto &rule : m_highlightingRules) {
-        // const QRegExp expression(rule.pattern);
-        // int index = text.indexOf(expression);
-        QRegExp expression(rule.pattern);
-        int index = expression.indexIn(text);
-        while (index >= 0) {
-            const int length = expression.matchedLength();
+        auto m = rule.pattern.match(text);
+        while (m.hasMatch()) {
+            int index = m.capturedStart();
+            const int length = m.capturedLength();
             setFormat(index, length, rule.format);
-            // index = text.indexOf(expression, index + length);
-            index = expression.indexIn(text, index + length);
+            m = rule.pattern.match(text, m.capturedEnd());
         }
```

Comment:
This could maybe work better with global matching?
https://doc.qt.io/qt-6/qregularexpression.html#global-matching